### PR TITLE
Feature ADF-1102 CI pipeline

### PR DESCRIPTION
**Pull request by wazelin**
# [ADF-1102](https://oat-sa.atlassian.net/browse/ADF-1102)

This PR aims to integrate the extension with a new CI pipeline defined by https://github.com/oat-sa/tao-extension-ci-action

⚠️ It is expected that PHP > 7.4 pipeline fail ⚠️

## Tasks
- [x] Revert https://github.com/oat-sa/tao-core/pull/3577/commits/c6a86a2579342823b47f0d0d8742ae7c26124ad3 once https://github.com/oat-sa/generis/pull/1018 has been merged and released

## Changelog
- [fix: avoid _exact_ version dependencies on semver-supporting packages](https://github.com/oat-sa/tao-core/commit/c85f2377297b1ee5ffbca731f0d043581573f0a8)
- [fix: `TaskLogTest`](https://github.com/oat-sa/tao-core/commit/7cac5e70b68cfb586308ccdb1766bded49efbe1c)
- [fix: `RdfValueCollectionRepositoryTest` test doubles' configurations](https://github.com/oat-sa/tao-core/commit/f4b8f6e49c4b3c88651cc03ff954faee25cf2792)
- [fix: `FormContainerTest `test doubles' configurations](https://github.com/oat-sa/tao-core/commit/f805ad0afdc48163006ac6e8ebbea211a20f7f52)
- [ci: add a CI action](https://github.com/oat-sa/tao-core/commit/337c21d47dda5450bf0bdbb96a130307eb22ea10)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -64,22 +64,22 @@
         "guzzlehttp/guzzle": "^6.5.7",
         "guzzlehttp/psr7": "^1.8.4",
         "imsglobal/lti": "^3.0",
-        "justinrainbow/json-schema": "5.2.1",
+        "justinrainbow/json-schema": "^5.2.1",
         "oat-sa/jig": "~0.1",
         "oat-sa/composer-npm-bridge": "~0.4.2||dev-master",
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
-        "oat-sa/generis": ">=15.20.3",
+        "oat-sa/generis": "dev-feature/ADF-1102/github-ci as 15.21.0",
```
Comment (kilatib) [inline composer.json:71]: Is it ok if it come to develop?

Comment (wazelin) [reply composer.json:71]: No, it's mentioned in the PR description.

Comment (kilatib) [reply composer.json:71]: sorry, yes missed it, I need a coffee before review anything :)

File: .github/workflows/continuous-integration.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Continuous integration
+
+on:
+  push:
+    branches: [ master, develop ]
+  pull_request:
+    branches: [ develop ]
```
Comment (kilatib) [inline .github/workflows/continuous-integration.yaml:7]: On some repos we don't have `develop`, but I see we have there `main` should we include it

Comment (wazelin) [reply .github/workflows/continuous-integration.yaml:7]: Not on the extensions, AFAIK.

And those rules apply only to this particular repo.

**Review by kilatib — APPROVED** [review]

**Review by gabrielfs7 — APPROVED** [review]

**Pull request merged**